compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp

```diff
@@ -0,0 +1,701 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FusePreActivationBatchNormPass.h"
+#include "FusePreActivationBatchNormPassInternal.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Log.h>
+
+#include <iostream>
+
+namespace
+{
+
+// Check if all elements are non-negative
+bool is_non_negative(luci::CircleConst *node)
+{
+  assert(node->dtype() == loco::DataType::FLOAT32);
+
+  uint32_t size = node->size<loco::DataType::FLOAT32>();
+  for (uint32_t i = 0; i < size; ++i)
+  {
+    if (node->at<loco::DataType::FLOAT32>(i) < 0)
+      return false;
+  }
+  return true;
+}
+
+// Check if mul is batchnorm mul
+bool is_batchnorm_mul(luci::CircleMul *mul, luci::CircleNode *&pred_node, luci::CircleConst *&gamma)
```

Comment:
same reasone, can we use `const luci::CircleMul *mul`

Reply:
Updated. Thanks.